x/axelarcork/keeper/keeper.go

```diff
@@ -197,15 +198,53 @@ func (k Keeper) GetScheduledAxelarCorksByID(ctx sdk.Context, chainID uint64, que
 // WinningCork //
 /////////////////
 
-func (k Keeper) SetWinningAxelarCork(ctx sdk.Context, chainID uint64, cork types.AxelarCork) {
+func (k Keeper) SetWinningAxelarCork(ctx sdk.Context, chainID uint64, blockHeight uint64, cork types.AxelarCork) {
 	bz := k.cdc.MustMarshal(&cork)
-	ctx.KVStore(k.storeKey).Set(types.GetWinningAxelarCorkKey(chainID, common.HexToAddress(cork.TargetContractAddress)), bz)
+	ctx.KVStore(k.storeKey).Set(types.GetWinningAxelarCorkKey(chainID, blockHeight, common.HexToAddress(cork.TargetContractAddress)), bz)
 }
 
-func (k Keeper) GetWinningAxelarCork(ctx sdk.Context, chainID uint64, contract common.Address) (types.AxelarCork, bool) {
+func (k Keeper) IterateWinningAxelarCorks(ctx sdk.Context, chainID uint64, cb func(contract common.Address, blockHeight uint64, cork types.AxelarCork) (stop bool)) {
 	store := ctx.KVStore(k.storeKey)
+	iter := sdk.KVStorePrefixIterator(store, types.GetWinningAxelarCorkKeyPrefix(chainID))
+	defer iter.Close()
+
+	for ; iter.Valid(); iter.Next() {
+		var cork types.AxelarCork
+		keyPair := bytes.NewBuffer(iter.Key())
+		keyPair.Next(1) // trim prefix byte
+		keyPair.Next(8) // trim chain ID
+		blockHeight := binary.BigEndian.Uint64(keyPair.Next(8))
```

Comment:
I understand this design was to not change the storage type, but given that we've landed on one enqueued call per contract address, is there a reason we couldn't switch this to an `AxelarScheduledCork` type and just have the block height stored with it, and do the lookup based on the chain ID and contract address?